src/main/java/mate.academy/lib/Injector.java

```
@@ -1,13 +1,77 @@
+                clazzImplementationInstance = createNewInstance(clazz);
+                try {
+                    field.setAccessible(true);
+                    field.set(clazzImplementationInstance,fieldInstance);
+                } catch (IllegalAccessException e) {
+                    throw new RuntimeException("Cant initialize field " + "Class: "
+                            + clazz.getName() + " field: " + field.getName());
+                }
+            }
+        }
+        if (clazzImplementationInstance == null) {
+            clazzImplementationInstance = createNewInstance(clazz);
+        }
+        return clazzImplementationInstance;
+    }
+
+    private Object createNewInstance(Class<?> clazz) {
+        if (instances.containsKey(clazz)) {
+            return instances.get(clazz);
+        }
+        try {
+            Constructor<?> constructor = clazz.getConstructor();
+            Object instance = constructor.newInstance();
+            instances.put(clazz, instance);
+            return instance;
+        } catch (NoSuchMethodException | IllegalAccessException | InstantiationException
+                 | InvocationTargetException e) {
+            throw new RuntimeException("Can't create a new instance " + clazz.getName());
+        }
+    }
+
+    private Class<?> findImplementation(Class<?> interfaceClazz) {
+        Map<Class<?>, Class<?>> interfaceImplementations = new HashMap<>();
+        Class<?> resultClazz;
+        interfaceImplementations.put(ProductParser.class, ProductParserImpl.class);
+        interfaceImplementations.put(FileReaderService.class, FileReaderServiceImpl.class);
+        interfaceImplementations.put(ProductService.class, ProductServiceImpl.class);
+        if (interfaceClazz.isInterface()) {
+            resultClazz = interfaceImplementations.get(interfaceClazz);
+        } else {
+            resultClazz = interfaceClazz;
+        }
+        if (resultClazz.isAnnotationPresent(Component.class)) {
```

flip the `if`, better to throw the exception inside the `if`'s body